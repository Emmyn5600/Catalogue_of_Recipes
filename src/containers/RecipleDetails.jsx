import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

const RecipeDetails = (props) => {
  const [selectedRecipe, setSelectedRecipe] = useState({});
  const { match, history, recipes } = props;
  const { params } = match;
  const { id } = params;
  useEffect(() => {
    if (!id) return history.replace('/');
    const recipe = recipes.find((r) => r.idMeal === id);
    if (!recipe) return history.replace('/');
    return setSelectedRecipe(recipe);
  }, []);

  const myIngredients = [];
  Object.keys(selectedRecipe).map((key, index) => (key.startsWith('strIngredient') && selectedRecipe[key] !== ''
    ? myIngredients.push({ name: selectedRecipe[key], id: index })
    : ''));
  return (
    <div>
      <div className="image-wrapper-1">
        <img src={selectedRecipe.strMealThumb} alt="recipe" />
        <div className="recipe-details">
          <h3 className="name">{selectedRecipe.strInstructions}</h3>
        </div>
        <div className="category-list">
          <h1>Ingredient</h1>
          <ul className="category-details">
            {myIngredients.map((ingredient) => (
              <li key={ingredient.id}>{ingredient.name}</li>
            ))}
          </ul>
        </div>
      </div>
    </div>
  );
};

RecipeDetails.defaultProps = {
  history: {},
  match: {
    params: { id: '11' },
  },
};

RecipeDetails.propTypes = {
  recipes: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
  history: PropTypes.objectOf(PropTypes.any),
  match: PropTypes.objectOf(PropTypes.any),
};

const mapStateToProps = (state) => ({
  recipes: state.recipes,
});

export default connect(mapStateToProps)(RecipeDetails);
