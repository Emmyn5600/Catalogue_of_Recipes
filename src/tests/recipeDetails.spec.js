import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';

import renderer from 'react-test-renderer';
import RecipleDetails from '../containers/RecipleDetails';
import createStore from '../store/createStore';

const store = createStore();

const recipe = {
  idMeal: '52977',
  strMeal: 'Corba',
  strDrinkAlternate: null,
  strCategory: 'Side',
  strArea: 'Turkish',
  strInstructions:
    'Pick through your lentils for any foreign debris, rinse them 2 or 3 times, drain, and set aside. Fair warning, this will probably turn your lentils into a solid block that you’ll have to break up later In a large pot over medium-high heat, sauté the olive oil and the onion with a pinch of salt for about 3 minutes, then add the carrots and cook for another 3 minutes. Add the tomato paste and stir it around for around 1 minute. Now add the cumin, paprika, mint, thyme, black pepper, and red pepper as quickly as you can and stir for 10 seconds to bloom the spices. Congratulate yourself on how amazing your house now smells. Immediately add the lentils, water, broth, and salt. Bring the soup to a (gentle) boil. After it has come to a boil, reduce heat to medium-low, cover the pot halfway, and cook for 15-20 minutes or until the lentils have fallen apart and the carrots are completely cooked. After the soup has cooked and the lentils are tender, blend the soup either in a blender or simply use a hand blender to reach the consistency you desire. Taste for seasoning and add more salt if necessary. Serve with crushed-up crackers, torn up bread, or something else to add some extra thickness. You could also use a traditional thickener (like cornstarch or flour), but I prefer to add crackers for some texture and saltiness. Makes great leftovers, stays good in the fridge for about a week.',
  strMealThumb:
    'https://www.themealdb.com/images/media/meals/58oia61564916529.jpg',
  strTags: 'Soup',
  strYoutube: 'https://www.youtube.com/watch?v=VVnZd8A84z4',
  strIngredient1: 'Lentils',
  strIngredient2: 'Onion',
  strIngredient3: 'Carrots',
  strIngredient4: 'Tomato Puree',
  strIngredient5: 'Cumin',
  strIngredient6: 'Paprika',
  strIngredient7: 'Mint',
  strIngredient8: 'Thyme',
  strIngredient9: 'Black Pepper',
  strIngredient10: 'Red Pepper Flakes',
  strIngredient11: 'Vegetable Stock',
  strIngredient12: 'Water',
  strIngredient13: 'Sea Salt',
};

describe('Recipe Details', () => {
  it('should render correctly', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Provider store={store}>
            <RecipleDetails recipe={recipe} />
          </Provider>
        </BrowserRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
