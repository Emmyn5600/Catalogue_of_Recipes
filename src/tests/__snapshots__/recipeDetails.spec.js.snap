// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recipe Details should render correctly 1`] = `
<div>
  <div
    className="image-wrapper-1"
  >
    <img
      alt="recipe"
    />
    <div
      className="recipe-details"
    >
      <h3
        className="name"
      />
    </div>
    <div
      className="category-list"
    >
      <h1>
        Ingredient
      </h1>
      <ul
        className="category-details"
      />
    </div>
  </div>
</div>
`;
